src/puzzles/day_8.rs

```diff
@@ -0,0 +1,108 @@
+use super::input;
+
+use self::computation::*;
+
+pub fn part_1() -> i32 {
+    let instructions: Vec<Instruction> = input::read_lines(8)
+        .iter()
+        .map(|line| Instruction::from(line).expect("Invalid instruction in input file src/inputs/day_8.txt. Get original file from https://adventofcode.com/2020/day/8/input"))
+        .collect();
+    let mut computer = Computer::from(instructions);
+    computer.run();
+    computer.accumulator
+}
+
+pub fn part_2() -> i32 {
+    0
+}
+
+mod computation {
+
+    use std::convert::TryFrom;
+
+    #[derive(Debug)]
+    pub enum Instruction {
+        Accumulate(i32),
+        Jump(i32),
+        NoOperation(i32),
+    }
+
+    impl Instruction {
+        pub fn from(instruction: &str) -> Result<Instruction, &str> {
+            match &instruction[..3] {
+                "acc" => match instruction[4..].parse::<i32>() {
+                    Ok(number) => return Ok(Instruction::Accumulate(number)),
+                    Err(_) => {
+                        return Err("Could not parse number from accumulate instruction.")
+                    }
+                },
+                "jmp" => match instruction[4..].parse::<i32>() {
+                    Ok(offset) => return Ok(Instruction::Jump(offset)),
+                    Err(_) => return Err("Could not parse offset from jump instruction."),
+                },
+                "nop" => match instruction[4..].parse::<i32>() {
+                    Ok(number) => return Ok(Instruction::NoOperation(number)),
+                    Err(_) => {
+                        return Err("Could not parse number from no operation instruction.")
+                    }
+                },
+                _ => return Err("Unknown instruction. Must be acc, jmp or nop"),
+            }
+        }
+    }
+
+    #[derive(Debug)]
+    pub struct Computer {
+        pub accumulator: i32,
```

Comment:
Remove `pub` and rather add a getter for the `accumulator` as it should not be mutable from outside.